src/api/server.rs

```diff
@@ -0,0 +1,134 @@
+use super::{handler, schema};
+use async_graphql::{
+    http::{playground_source, GraphQLPlaygroundConfig},
+    QueryBuilder,
+};
+use async_graphql_warp::{graphql_subscription, GQLResponse};
+use std::convert::Infallible;
+use std::net::{IpAddr, SocketAddr};
+use tokio::sync::oneshot;
+use warp::{http::Response, Filter};
+
+pub struct Server {
+    /// Address for the API server to bind on
+    address: SocketAddr,
+
+    /// Transmission channel to trigger closure of a running API server
+    trigger_cancel: oneshot::Sender<()>,
+
+    /// Receiver signal to cancel running API server
+    cancel_signal: Option<oneshot::Receiver<()>>,
+
+    /// Enables the playground
+    playground: bool,
+}
+
+impl Server {
+    /// Returns a new API Server
+    pub fn new(address: SocketAddr) -> Self {
+        let (trigger_cancel, cancel_signal) = oneshot::channel::<()>();
+
+        Server {
+            address,
+            trigger_cancel,
+            cancel_signal: Some(cancel_signal),
+            playground: true,
+        }
+    }
+
+    pub fn set_playground(mut self, enable: bool) -> Self {
+        self.playground = enable;
```

Comment:
Consider using https://docs.rs/getset/0.1.1/getset/index.html and https://docs.rs/derive_builder/0.9.0/derive_builder/